OpenRA.Mods.Common/MapGenerator/Direction.cs

```diff
@@ -93,21 +102,21 @@ public static class Direction
 			Spread4.Select(xy => new CVec(xy.X, xy.Y)).ToImmutableArray();
 
 		/// <summary>Adjacent offsets with directions, including diagonals.</summary>
-		public static readonly ImmutableArray<(int2, int)> Spread8D =
+		public static readonly ImmutableArray<(int2, Direction)> Spread8D =
```

Comment:
Making this an `ImmutableDictionary<Direction, int2>` would simplify the code that needs to index it.

Reply:
Generally you'd use touples when you don't need to index into the collection. I don't see that we need indexing anywhere?

Reply:
The indexing is into `Spread8`/`Spread8CVec` which would be removed (loops could use `Spread8D.Values`).

Reply:
Loops are a main user of these arrays. I don't think reworking the various loops where it's used is ideal. I'm not convinced that it significantly simplifies code. One could argue it might even make it slightly more complex?

There's also some technicalities in that ImmutableDictionary doesn't officially provide ordering guarantees. That alone makes code harder for a casual reader to reason about.

ImmutableArray seems like the best choice in my opinion.